openquake/calculators/disaggregation.py

```diff
@@ -191,9 +201,14 @@ def full_disaggregation(self):
                 int(numpy.ceil(max_mag / mag_bin_width) + 1))
             logging.info('%d mag bins from %s to %s', len(mag_edges) - 1,
                          min_mag, max_mag)
-            for sid, site in zip(sitecol.sids, sitecol):
-                curves = curves_dict[sid]
-                if not curves:
+            for i, site in enumerate(sitecol):
+                sid = sitecol.sids[i]
+                curve = curves[i]
+                for rlzi, imt in curve:
+                    maxc = curve[rlzi, imt].max()
```

Comment:
I think lines 208-211 can be also written as
max_poe[rlzi][imt] = max(max_poe[rlzi][imt], curve[rlzi, imt].max())

Reply:
Done